src/generic/configure-modal/ConfigureModal.test.jsx

```diff
@@ -390,16 +390,152 @@ describe('<ConfigureModal /> for Unit', () => {
   });
 
   it('disables the Save button and enables it if there is a change', () => {
-    useSelector.mockReturnValue(
+    const newCurrentItemData = {
+      ...currentUnitMock,
+      userPartitionInfo: {
+        ...currentUnitMock.userPartitionInfo,
+        selectedPartitionIndex: 0,
+      },
+    };
+    const { getByRole, getByTestId } = renderUnitComponent({
+      currentItemData: newCurrentItemData,
+    });
+
+    const saveButton = getByRole('button', { name: messages.saveButton.defaultMessage });
+    expect(saveButton).toBeDisabled();
+
+    const input = getByTestId('group-type-select');
+    // unrestrict access
+    fireEvent.change(input, { target: { value: -1 } });
+    expect(saveButton).not.toBeDisabled();
+
+    fireEvent.change(input, { target: { value: 0 } });
+    expect(saveButton).toBeDisabled();
+
+    const checkbox = getByTestId('unit-visibility-checkbox');
+    fireEvent.click(checkbox);
+    expect(saveButton).not.toBeDisabled();
+  });
+});
+
+const currentXBlockMock = {
```

Comment:
Can we move mocks to separate mock file. They take too much space in the file.

Reply:
Refactored